newman-tests/data-browser/data-browser-api-test.json

```diff
@@ -0,0 +1,483 @@
+          "let expectedResponse = JSON.stringify(testData[testNumber]? testData[testNumber].expectedResponse:{} )",
+          "let sourceFile = testData[testNumber].sourceFile",
+          "++testNumber",
+          "pm.variables.set(\"testNumber\",testNumber);",
+          "pm.variables.set(\"requestBody\",requestBody);",
+          "pm.variables.set(\"sourceFile\",sourceFile);"
+        ]
+      }
+    }
+  ],
+  "item": [
+    {
+      "name": "Data Browser API Health Check",
+      "event": [
+        {
+          "listen": "test",
+          "script": {
+            "type": "text/javascript",
+            "exec": [
+              "pm.test(\"Status code is 404\",function () {",
+              "      pm.response.to.have.status(404);",
+              "});",
+              ""            ]
+          }
+        }
+      ],
+      "request": {
+        "url": "http://{{host-data-browser-api}}",
+        "method": "GET",
+        "header": [
+          {
+            "key": "Content-Type",
+            "value": "application/json"
+          }
+        ],
+        "body": {},
+        "description": "Data Browser API HEALTH CHECK"
+      },
+      "response": []
+    },
+    {
+      "name": "Data Browser Aggregation End Point Check",
+      "event": [
+        {
+          "listen": "test",
+          "script": {
+            "type": "text/javascript",
+            "exec": [
+              "pm.test(\"Status code is 200\",function () {",
+              "      pm.response.to.have.status(200);",
+              "});",
+              "",
+              "pm.test(\"GET Aggregation\", function () {",
+              "let actualResponse = pm.response.json()",
+              "let testData = pm.iterationData.get(\"testData\")",
+              "let testDataChunk = testData[testNumber-1]",
+							"console.log('Full response: ' + JSON.stringify(actualResponse))",
+              "console.log(JSON.stringify(actualResponse['parameters']))",
+							"console.log(JSON.stringify(testDataChunk.expectedResponse))",
+              "pm.expect(actualResponse['parameters']).to.eql(testDataChunk.expectedResponse)",
+              "});"            ]
+          }
+        }
+      ],
+      "request": {
+        "url":  "http://{{host-data-browser-api}}/view/aggregations?{{test-data-1}}",
+        "method": "GET",
+        "header": [
+          {
+            "key": "Content-Type",
+            "value": "application/json"
+          }
+        ],
+        "body": {},
+        "description": "Data Browser Aggregation Check"
+      },
+      "response": []
+    },
+    {
+      "name": "Data Browser CSV End Point Check",
+      "event": [
+        {
+          "listen": "test",
+          "script": {
+            "type": "text/javascript",
+            "exec": [
+              "pm.test(\"Status code is 200\",function () {",
+              "      pm.response.to.have.status(200);",
```

Comment:
There should be a method here to validate the CSV being returned. I've experimented a bit with this here  for the rate spread Newman tests :

https://github.cfpb.gov/HMDA-Operations/hmda-devops/tree/master/ratespread-test

Reply:
We can't verify the data returned by CSV or PIPE because the data returned is unique in each environment and will change each year.